lib/astutils.cpp

```diff
@@ -1547,6 +1547,18 @@ static bool isUnchanged(const Token *startToken, const Token *endToken, const st
     return true;
 }
 
+bool isNullOperand(const Token *expr)
+{
+    if (!expr)
+        return false;
+    if (Token::Match(expr, "( %name% %name%| * )") &&
```

Comment:
Maybe we could handle c++ casts better.

Something like:

    if (!expr->isCast())
        return Token::Match(expr, "NULL|nullptr");
    if (expr->valueType() && expr->valueType()->pointer == 0)
        return false;
    const Token *castOp = expr->astOperand2() ? expr->astOperand2() : expr->astOperand1();
    return Token::Match(castOp, "NULL|nullptr") || (MathLib::isInt(castOp->str()) && MathLib::isNullValue(castOp->str()));

Reply:
I didn't get `isCast()` true for any tokens in `static_cast<int*>(0)` (is this the expected outcome?) so I hade to add an explicit check for the c++ casts. Perhaps there should be a helper for this, it seems like it could be used in many places?

Reply:
strange. The `isCast()` should be true for the `(` token. We can fix that separately.

Reply:
> strange. The isCast() should be true for the ( token. We can fix that separately.

I can reproduce this. It feels like a bug. Maybe there is something I am missing.